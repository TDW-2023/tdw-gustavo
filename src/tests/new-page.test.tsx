import React from "react";
import { render, screen } from "@testing-library/react";
import renderer from "react-test-renderer";
import NewPage from "../../pages/new-page";

test('Text Validation: "Fortaleza"', () => {
  render(<NewPage />);
  const textElements = screen.getAllByText(/Fortaleza/i);
  expect(textElements.length).toBeGreaterThanOrEqual(0);

  const numberOfOccurrences: number = textElements.length;
  expect(numberOfOccurrences).toBeGreaterThan(0);
  console.log(
    `Número de ocorrências da palavra "Fortaleza": ${numberOfOccurrences}`,
  );
});

test("Snapshot for NewPage", () => {
  const component = renderer.create(<NewPage />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
