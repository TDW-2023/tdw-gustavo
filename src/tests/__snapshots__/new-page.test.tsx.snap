// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot for NewPage 1`] = `
<main
  className="fortalezaPage"
>
  <header>
    <h1>
      Fortaleza Esporte Clube
    </h1>
    <img
      alt="Escudo do Fortaleza"
      className="vercelLogo"
      data-nimg="1"
      decoding="async"
      fetchpriority="high"
      height={100}
      onError={[Function]}
      onLoad={[Function]}
      src="/_next/image?url=%2Ffortaleza.png&w=256&q=75"
      srcSet="/_next/image?url=%2Ffortaleza.png&w=128&q=75 1x, /_next/image?url=%2Ffortaleza.png&w=256&q=75 2x"
      style={
        {
          "color": "transparent",
        }
      }
      width={100}
    />
  </header>
  <section>
    <h2>
      Sobre o Fortaleza
    </h2>
    <p>
      O Fortaleza Esporte Clube é um clube de futebol brasileiro sediado em Fortaleza, no estado do Ceará. Fundado em 1918, é um dos clubes mais tradicionais do Nordeste.
    </p>
  </section>
  <section>
    <h2>
      Principais Conquistas
    </h2>
    <ul
      className="fortalezaConquers"
    >
      <li>
        Campeonato Brasileiro Série B: 2018
      </li>
      <li>
        Copa do Nordeste: 1960, 1964, 1966, 1971, 2000, 2019
      </li>
      <li>
        Copa dos Campeões Cearenses: 2002
      </li>
    </ul>
  </section>
  <section>
    <h2>
      Estádio
    </h2>
    <p>
      O Fortaleza manda seus jogos no estádio Castelão, com capacidade para aproximadamente 63.903 torcedores.
    </p>
  </section>
  <footer
    className="fortalezaFooter"
  >
    <p>
      Para mais informações, visite o site oficial do
       
      <a
        href="https://www.fortalezaec.net/"
        target="_blank"
      >
        Fortaleza Esporte Clube
      </a>
      .
    </p>
  </footer>
</main>
`;
